src/natcap/invest/utils.py

```diff
@@ -69,34 +69,41 @@ def _log_gdal_errors(*args, **kwargs):
     Returns:
         ``None``
     """
-    if len(args) + len(kwargs) != 3:
-        LOGGER.error(
-            '_log_gdal_errors was called with an incorrect number of '
-            f'arguments.  args: {args}, kwargs: {kwargs}')
-
     try:
-        gdal_args = {}
-        for index, key in enumerate(('err_level', 'err_no', 'err_msg')):
+        # We had a case on the forums where GDAL was calling the error handler
+        # with no args or kwargs at all - it isn't even clear why the error
+        # handler was called in the first place..  Handling it here at the
+        # DEBUG level to avoid exceptions being logged to the logfile.
+        if len(args) + len(kwargs) == 0:
+            LOGGER.debug("_log_gdal_errors was called with no args/kwargs. "
+                         "Skipping.")
+            return
+    except Exception as e:
+        LOGGER.exception(str(e))
```

Comment:
How do we get here, some improperly typed `args` or `kwargs`? If so, do we want to proceed to with logging, or should we `return/raise` here?

Reply:
It is super unclear because I'm having a tough time reproducing it on my system, but the user I'm working with on the forums was seeing something in `len` raising a `UnicodeDecodeError`.  And because of the nature of this error, I'm having a really hard time actually logging the object itself, or even the type!  It's very bizarre.

I added a note about this in c883dc196 and am now returning from the logging function.  You're right, if we're seeing this same error, we probably won't be able to do much about the rest of the logging.